dart/2.Dart_basics/Excercises/4_solution.md

```diff
@@ -0,0 +1,30 @@
+# Solution 4
+
+```dart
+void main() {
+  String message = ("This is a message. And I will replace it's vowels");
+  String newMessage = message.replaceAll('a', '*');
+  newMessage = newMessage.replaceAll('e', '*');
+  newMessage = newMessage.replaceAll('i', '*');
+  newMessage = newMessage.replaceAll('o', '*');
+  newMessage = newMessage.replaceAll('u', '*');
+  newMessage = newMessage.replaceAll('A', '*');
+  newMessage = newMessage.replaceAll('E', '*');
+  newMessage = newMessage.replaceAll('I', '*');
+  newMessage = newMessage.replaceAll('O', '*');
+  newMessage = newMessage.replaceAll('U', '*');
+  print(newMessage);
+}
+```
+
+Although this code solves the exercise, we are not complying with the DRY rule (we'll talk about it in section 4).
+
+The following code does the same but in a much more efficient way. Here we can see that a new concept is introduced: `RegExp()` to solve it. Of course you already know what to do: go read the documentation about it!
```

Comment:
I think it's super cool to do this kind of exercise but we should create an article to explain how to use RegEx!

Reply:
Ok! I'll work on it, where do you think it should be? I believe it doesn't fit in any section. Maybe create a new section called 'Interesting commands' or something like that where we add specific articles about this kind of commands and the in the exercise put a likn to it.

Reply:
You know that we have or we'll have, a part of our web which is curated learning courses, meaning a list of courses in order to learn. Other place, is isolated articles. In that part I think should go.

Reply:
(Feel free to resolve this - I've left it for you to read the comment)